project.php

```diff
@@ -248,6 +248,43 @@ function decide_blurbs()
     if ( $blurb )
         return array( $blurb, $blurb );
 
+    // Check whether the user is blocked by a daily page limit.
+    // Arguably, you'd expect this to be determined in can_user_get_pages_in_project(),
+    // but we also want to produce a warning (when the project is covered by a DPL)
+    // even when the user *isn't* blocked, which you can't really expect of that function.
+    //
+    $blocked_by_limit_message = "";
+    $page_limit_warning = "";
+    if ($round->has_a_daily_page_limit())
+    {
+        $user_dpl_count = $round->get_dpl_count_for_user($pguser);
+        if ($user_dpl_count >= $round->daily_page_limit)
+        {
+            // User has reached this round's DPL.
+            $blocked_by_limit_message = sprintf(
+                _("This project contributes to the '%s' limit of %d page-saves per day, and you have already reached that limit today."),
+                $round->id,
+                $round->daily_page_limit
+            );
+        }
+        else
+        {
+            // User hasn't reached this round's DPL,
+            // but they should be warned that there *is* a DPL for this round?
+            $msg = sprintf(
+                _("Warning: This project contributes to the '%s' limit of %d page-saves per day. Since server midnight, your current count is <b>%d</b>."),
```

Comment:
Perhaps:
```
This project contributes to the %s limit of %d page-saves per day since server midnight. Your current count is <b>%d</b>.
```
If it's in the warning CSS class it shouldn't need the label and I don't think we need the single quotes around the round.

Reply:
I've changed it to @srjfoo's wording.

I also changed the other message to eliminate the "This project contributes to" phrase; have a look.

Reply:
I kept the "<b>" for the user's current dpl-count, but note that it has no effect, because layout.css says that 'warning' has font-weight bold.

Reply:
This LGTM!